src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs

```diff
@@ -59,56 +105,91 @@ private TocItemInfo ResolveItemCore(TocItemInfo wrapper, Stack<FileAndType> stac
                         {
                             item.Items[i] = ResolveItem(new TocItemInfo(file, item.Items[i]), stack).Content;
                         }
-                        if (string.IsNullOrEmpty(item.Homepage) && string.IsNullOrEmpty(item.HomepageUid))
+                        if (string.IsNullOrEmpty(item.TopicHref) && string.IsNullOrEmpty(item.TopicUid))
                         {
                             var defaultItem = GetDefaultHomepageItem(item);
                             if (defaultItem != null)
                             {
-                                item.Homepage = defaultItem.Href;
-                                item.HomepageUid = defaultItem.Uid;
+                                item.AggregatedHref = defaultItem.Href;
+                                item.AggregatedUid = defaultItem.TopicUid;
                             }
                         }
                     }
-
+                    // for backward compatibility
+                    if (item.Href == null && item.Homepage == null)
+                    {
+                        item.Href = item.TocHref;
+                        item.Homepage = item.TopicHref;
+                    }
+                    // check whether toc exists
+                    if (!string.IsNullOrEmpty(item.TocHref) &&
+                        (tocHrefType == HrefType.MarkdownTocFile || tocHrefType == HrefType.YamlTocFile))
+                    {
+                        var tocFilePath = (RelativePath)file.File + (RelativePath)item.TocHref;
+                        var tocFile = new FileAndType(file.BaseDir, tocFilePath, file.Type, file.PathRewriter);
+                        if (!_collection.ContainsKey(tocFile))
```

Comment:
Move this check out side the switch, This should always happen even if href is empty.